test/e2e/utils/test_context.go

```diff
@@ -50,36 +50,30 @@ func NewTestContext(binaryName string, env ...string) (*TestContext, error) {
 		return nil, err
 	}
 
-	cc := &CmdContext{
-		Env: env,
-	}
-
-	// Use kubectl to get Kubernetes client and cluster version.
-	kubectl := &Kubectl{
-		Namespace:  fmt.Sprintf("e2e-%s-system", testSuffix),
-		CmdContext: cc,
-	}
-	k8sVersion, err := kubectl.Version()
+	testGroup := "bar" + testSuffix
+	path, err := filepath.Abs("e2e-" + testSuffix)
 	if err != nil {
 		return nil, err
 	}
 
-	// Set CmdContext.Dir after running Kubectl.Version() because dir does not exist yet.
-	if cc.Dir, err = filepath.Abs("e2e-" + testSuffix); err != nil {
-		return nil, err
+	cc := &CmdContext{
+		Env: env,
+		Dir: path,
 	}
 
 	return &TestContext{
 		TestSuffix: testSuffix,
 		Domain:     "example.com" + testSuffix,
-		Group:      "bar" + testSuffix,
+		Group:      testGroup,
 		Version:    "v1alpha1",
 		Kind:       "Foo" + testSuffix,
 		Resources:  "foo" + testSuffix + "s",
 		ImageName:  "e2e-test/controller-manager:" + testSuffix,
 		CmdContext: cc,
-		Kubectl:    kubectl,
-		K8sVersion: &k8sVersion,
+		Kubectl: &Kubectl{
+			Namespace:  fmt.Sprintf("e2e-%s-system", testSuffix),
+			CmdContext: cc,
+		},
 		BinaryName: binaryName,
```

Comment:
Just revert the change made in: https://github.com/kubernetes-sigs/kubebuilder/commit/ea3a94a97ad4ba0cb148e711095530286d050d54#diff-7b614da29899d7d7eb4aa3b60ed598604da518c4f2dd1e6af954c942fb1c3dc3 and keep the k8sVersion